.github/workflows/ci.yml

```diff
@@ -4,36 +4,37 @@ name: Build and Deploy Documentation using MkDocs
 # but pull request events on any branch but deployement happens only for the main branch
 on:
   push:
-    branches:
-      - main
+    branches: [main]
   pull_request:
-  #   branches: [main]
+  #  branches: [main]
 
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
+# permissions:
+#   contents: read
+
 jobs:
   build:
     name: Build Documentation
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: [3.10.12]
 
     steps:
       - name: Checkout Main
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-          architecture: x64
       - name: Install dependencies and setup git config
         run: |
           python -m pip install --upgrade pip
-          pip install -r requirements.txt
-          git config user.name 'github-actions[bot]' && git config user.email 'github-actions[bot]@users.noreply.github.com'
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          git config user.name 'github-actions[bot]' && git config user.email 'help@nerc.mghpcc.org'
```

Comment:
are we sure we want to add help@nerc.mghpcc.org here?  what are the consequences?

Reply:
> Using the email address 'github-actions[bot]@users.noreply.github.com' for automated processes like GitHub Actions is a common practice. It ensures that automated commits are clearly associated with the bot account and don’t clutter your personal email inbox.

I will revert it back.